src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,67 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints(dist.entry_points)
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    @staticmethod
+    def _parse_entrypoint(entrypoint: str) -> Tuple[str, str]:
+        """
+        Extract the entry point and name from the string.
+
+        .. code-block:: python
+
+            WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
+            # returns ('hello-world', 'timmins:hello_world')
+
+            WritePyproject._parse_entrypoint("hello-world")
+            # throws ValueError: Entry point 'hello-world' is not of the form 'name = target'
+
+        :param: entrypoint  The entry point string, of the form
+                            "entrypoint = module:function" (whitespace optional)
+        :returns:           A two-element `tuple`, first element is the entry point name, second element is the target
+                            (module and function name) as a string.
+        :raises ValueError: An equals (`=`) character was not present in the entry point string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
+        if "=" not in entrypoint:
+            raise ValueError("Entry point %r is not of the form 'name = module:function'" % entrypoint)
+
+        (name, target) = entrypoint.split("=", 1)
+        return (name.strip(), target.strip())
+
+    @classmethod
+    def _generate_entrypoints(cls, entrypoints: Optional[Dict[str, List[str]]]) -> Dict[str, Dict[str, str]]:
+        """
+        Dump the entry-points given, if any.
```

Comment:
I think it'd help a bit to be consistent about whether it's spelled "entrypoints", "entry-points", or "entry points", although I'm not sure which one is best. (TBH I'm not even sure if setuptools itself is consistent about this... if not, I guess this comment becomes kind of moot)

Reply:
Yeah, not completely sure myself.  The property on the `Distribution` object is called `entry_points`; whether you read the `_` as being a space or an underscore is up for debate.

In the docs, it's not hyphenated in the title, it's a space: https://setuptools.pypa.io/en/latest/userguide/entry_point.html -- so yeah, maybe the hyphen can go.

Reply:
In that case I'd go with "entry points" for docs and `entry_points` for code.

Reply:
Yep, I've just revised everything so we're using `entry_points` in code, "entry points" in documentation.  The only place where we use `entry-points` is in the TOML output.